### ISSUE TITLE
Error duplicated torrents

### ISSUE THREAD

**Issue by IngenieroFiestero** [feross/webtorrent#273]
events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: Swarm listen error: There is already a swarm with infoHash 19a6ce37f136e6fda1f0e18efee768c42ec38f6a listening on port 33294

**Comment by feross** [comment]
Please share the code you wrote that caused this error.

**Comment by IngenieroFiestero** [comment]
i'm working in a simple torrent client for raspberry pi, but i have a problem when i try to put the same torrent twice

**Comment by astro** [comment]
Very off-topic but I'm interested in targeting the Raspberry Pi for the following reasons:
* Due to the ongoing hype, there are now millions of unused Pis on this planet.
* Specialized appliances let you dictate the platform, allowing to care just about Linux.
* Low-power devices save the energy bill when seeding longer.

Is this going to be a client with a native front-end like Popcorn-Time? I recently created a webtorrent branch there. Or is it going to be a web app to be used by a remote computer? Perhaps you can help me developing [Bitorama](https://github.com/astro/bitorama)

**Comment by IngenieroFiestero** [comment]
In my apartament needed a server to download torrent files to not have all
computers turned on while consuming bandwidth and electricity. The best way
was to use a raspberry pi does not consume a lot of energy and allows
access from the browser to download files from any device connected to the
network.
It would be interesting accessed from anywhere. Eg download a file from the
bar, whe u get home is fully downloaded.
Sorry but im trying to improove my english.

**Comment by astro** [comment]
gustaria que tu pruebas Bitorama :-)

@feross The code is probably at https://github.com/IngenieroFiestero/piTorrent

**Comment by feross** [comment]
@IngenieroFiestero You can't add the same torrent to the client twice – it's forbidden since it makes no sense. What are you trying to do?

**Comment by IngenieroFiestero** [comment]
If a client or two clients adds the same torrent, the server crashes. Im
searching bugs.

**Comment by Krizz** [comment]
There should be a way to handle those kind of errors more elegantly when they happen. Something like...
client.add(magnetUri, function (err, torrent) {
  if (err) {
    //handle error
  }
});

**Comment by feross** [comment]
You can detect errors on the webtorrent client by adding an `'error'` handler:

```js
client.on('error', function (err) {
  console.error(err.stack)
})
```

**Comment by feross** [comment]
@IngenieroFiestero This issue is fixed in `bittorrent-swarm` v2.0.1.
